components/org.wso2.carbon.identity.recovery/src/test/java/org/wso2/carbon/identity/recovery/connector/AdminForcedPasswordResetConfigImplTest.java

```diff
@@ -0,0 +1,149 @@
+
+import org.apache.commons.lang.StringUtils;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryConstants;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.testng.Assert.assertEquals;
+
+public class AdminForcedPasswordResetConfigImplTest {
+    private AdminForcedPasswordResetConfigImpl adminForcedPasswordResetConfigIml;
+
+    @BeforeTest
+    public void init() {
+
+        adminForcedPasswordResetConfigIml = new AdminForcedPasswordResetConfigImpl();
+
+    }
+
+    @Test
+    public void testGetName() throws Exception {
+    }
+
+    @Test
+    public void testGetFriendlyName() throws Exception {
+    }
+
+    @Test
+    public void testGetCategory() throws Exception {
+    }
+
+    @Test
+    public void testGetSubCategory() throws Exception {
+    }
+
+    @Test
+    public void testGetOrder() throws Exception {
+    }
+
+    @Test
+    public void testGetPropertyNameMapping() throws Exception {
+        Map<String, String> nameMapping = new HashMap<>();
+        nameMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_WITH_RECOVERY_LINK,
+                "Enable Password Reset via Recovery Email");
+        nameMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_WITH_OTP,
+                "Enable Password Reset via OTP");
+        nameMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_OFFLINE,
+                "Enable Password Reset Offline");
+
+
+        Map<String, String> nameMappingExpected = adminForcedPasswordResetConfigIml.getPropertyNameMapping();
+        assertEquals(nameMapping, nameMappingExpected, "Maps are equal");
+
+
+    }
+
+    @Test
+    public void testGetPropertyDescriptionMapping() throws Exception {
+        Map<String, String> descriptionMapping = new HashMap<>();
+        descriptionMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_WITH_RECOVERY_LINK,
+                "User gets notified with a link to reset password");
+        descriptionMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_WITH_OTP,
+                "User gets notified with a one time password to try with SSO login");
+        descriptionMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_ADMIN_PASSWORD_RESET_OFFLINE,
+                "An OTP generated and stored in users claims");
+
+        Map<String, String> descriptionMappingExpected = adminForcedPasswordResetConfigIml.getPropertyDescriptionMapping();
+        assertEquals(descriptionMapping, descriptionMappingExpected, "Maps are equal");
+    }
+
+    @Test
+    public void testGetPropertyNames() throws Exception {
```

Comment:
do not throw generic exceptions

Reply:
Ok. I'll use IdentityGovernanceException.